lms/static/js/header/header.js

```diff
@@ -0,0 +1,172 @@
+    var $userMenu = $('.user-dropdown');
+    var $hamburgerMenu = $('.hamburger-menu');
+    var $toggleUserDropdown = $('.toggle-user-dropdown');
+
+    // Open or close relevant menu on enter or space click and focus on first element.
+    if ((e.keyCode === 13 || e.keyCode === 32) && (isToggle || isHamburgerMenu)) {
+        $(e.target).click();
+        if (isHamburgerMenu) {
+            if ($('.mobile-menu').is(':visible')) {
+                $hamburgerMenu.attr('aria-expanded', true);
+                $('.mobile-menu .mobile-nav-link a').first().focus();
+            } else {
+                $hamburgerMenu.attr('aria-expanded', false);
+            }
+        } else if (isToggle) {
+            if ($('.global-header .nav-item .dropdown-user-menu').is(':visible')) {
+                $userMenu.attr('aria-expanded', 'true');
+                $('.dropdown-item a:first').focus();
+            } else {
+                $userMenu.attr('aria-expanded', false);
+            }
+        }
+        // Don't allow for double click or page jump on Firefox browser
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    // Enable arrow functionality within the menu.
+    if (e.keyCode === 38 || e.keyCode === 40 && (isDropdownOption || isMobileOption ||
+        (isHamburgerMenu && $hamburgerMenu.hasClass('open')) || isToggle && $toggleUserDropdown.hasClass('open'))) {
+        isNext = e.keyCode === 40;
+        if (isNext && !isHamburgerMenu && !isToggle && isLastItem) {
+            // Loop to the start from the final element
+            nextLink = isDropdownOption ? $toggleUserDropdown : $hamburgerMenu;
+        } else if (!isNext && (isHamburgerMenu || isToggle)) {
+            // Loop to the end when up arrow pressed from menu icon
+            nextLink = isHamburgerMenu ? $('.mobile-menu .mobile-nav-link a').last()
+                : $('.dropdown-user-menu .dropdown-nav-item').last().find('a');
+        } else if (isNext && (isHamburgerMenu || isToggle)) {
+            // Loop to the first element from the menu icon
+            nextLink = isHamburgerMenu ? $('.mobile-menu .mobile-nav-link a').first()
+                : $('.dropdown-user-menu .dropdown-nav-item').first().find('a');
+        } else {
+            // Loop up to the menu icon if first element in menu
+            if (!isNext && $(e.target).parent().is(':first-child') && !isHamburgerMenu && !isToggle) {
+                nextLink = isDropdownOption ? $toggleUserDropdown : $hamburgerMenu;
+            } else {
+                nextLink = isNext ?
+                    $(e.target).parent().next().find('a') : // eslint-disable-line newline-per-chained-call
+                    $(e.target).parent().prev().find('a'); // eslint-disable-line newline-per-chained-call
+            }
+        }
+        nextLink.focus();
+
+        // Don't let the screen scroll on navigation
+        e.preventDefault();
+        e.stopPropagation();
```

Comment:
Move these to the top of the if statement wrapper, this may be what's causing the inconsistent behaviour.